### PR TITLE
Move some user routers to participant/user routes

### PR DESCRIPTION
**Pull request by alex-yau-ttd**
- Change a few user routes to be participantUser routes (of the form `/participants/:participantId/users/:userId`). Move these from usersRouter to participantUsersRouter - add this as a subrouter to the participantsRouter. Actual functionality of the endpoints is unchanged
- Pass participantId from front-end using `ParticipantContext`
- Remove an unused function & endpoint

## Manual testing
Perform resend invitation or delete/update user and notice the request URL is now of the form `http://localhost:3000/api/participants/8/users/7181` rather than `http://localhost:3000/api/users/7181`

### CODE REVIEW

File: src/api/routers/participantUsersRouter.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import express from 'express';
+
+import { UserController } from '../controllers/userController';
+import { enrichWithUserFromParams } from '../middleware/usersMiddleware';
+import { LoggerService } from '../services/loggerService';
+import { UserService } from '../services/userService';
+
+const createParticipantUsersRouter = () => {
+  const participantUsersRouter = express.Router({ mergeParams: true });
```
Comment (alex-yau-ttd) [inline src/api/routers/participantUsersRouter.ts:9]: `mergeParams` allows this router to access the params from the parent router, in this case, `participantId`.

File: src/api/controllers/userController.ts [hunk 2]
```diff
@@ -90,11 +90,6 @@ export class UserController {
     res.sendStatus(200);
   }
 
-  @httpGet('/:userId')
```
Comment (alex-yau-ttd) [inline src/api/controllers/userController.ts:93]: This was unused

File: src/api/routers/usersRouter.ts [hunk 3]
```diff
@@ -20,14 +20,6 @@ const createUsersRouter = () => {
     '/selfResendInvitation',
     userController.selfResendInvitation.bind(userController)
   );
-  usersRouter.use('/:userId', enrichWithUserFromParams);
-  usersRouter.get('/:userId', userController.getUserById.bind(userController));
```
Comment (alex-yau-ttd) [inline src/api/routers/usersRouter.ts:24]: This was unused. The other endpoints were just moved to the `participantUsersRouter`

File: src/web/services/userAccount.ts [hunk 4]
```diff
@@ -23,16 +23,6 @@ export type UpdateTeamMemberForm = Omit<InviteTeamMemberForm, 'email'>;
 
 export type UserPayload = z.infer<typeof UserCreationPartial>;
 export type UserResponse = UserDTO;
-export async function GetUserAccountById(id: string) {
```
Comment (alex-yau-ttd) [inline src/web/services/userAccount.ts:26]: This was unused

File: src/web/services/userAccount.ts [hunk 5]
```diff
@@ -75,30 +65,31 @@ export async function SelfResendInvitation(formData: SelfResendInvitationForm):
 
 export async function GetAllUsersOfParticipant(participantId: number) {
   try {
-    const result = await axios.get<UserResponse[]>(
-      `/participants/${participantId}/users`,
-      {
-        validateStatus: (status) => [200, 404].includes(status),
-      }
-    );
+    const result = await axios.get<UserResponse[]>(`/participants/${participantId}/users`, {
```
Comment (alex-yau-ttd) [inline src/web/services/userAccount.ts:68]: Formatting only

File: src/api/routers/participantUsersRouter.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const userController = new UserController(new UserService(), new LoggerService());
```
Comment (alex-yau-ttd) [inline src/api/routers/participantUsersRouter.ts:10]: I thought about creating a new controller & service but I figured it probably wasn't worth it at this stage, given we may change our inversify approach for dependency injection anyway.

File: src/web/services/userAccount.ts [hunk 7]
```diff
@@ -46,9 +36,9 @@ export async function GetLoggedInUserAccount(): Promise<UserWithIsApprover | nul
   }
 }
 
-export async function ResendInvite(id: number): Promise<void> {
+export async function ResendInvite(id: number, participantId: number): Promise<void> {
```
Comment (alex-yau-ttd) [inline src/web/services/userAccount.ts:39]: This function doesn't strictly need `participantId` for it to work, but I figured it still be useful for audit trail purposes.

**Review by ssundahlTTD — APPROVED** [review]

**Pull request merged**